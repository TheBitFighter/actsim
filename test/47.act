import std;
open std::channel;

defproc src (a1of2! x)
{
  int i;
  chp {
     i := 0;
    *[ i < 100 -> x!(i & 1); i := i + 1; log("sent!") ]
  }
}

defproc sink (a1of2? x)
{
  bool ax, bx;
  prs {
    x.t => ax+
    x.f => bx+
    ax | bx => x.a+
  }
}

defproc test ()
{
   src s;
   sink t(s.x);
}
