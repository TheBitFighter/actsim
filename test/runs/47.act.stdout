[                 110] <s>  sent!
[                 190] <s>  sent!
[                 270] <s>  sent!
[                 350] <s>  sent!
[                 430] <s>  sent!
[                 510] <s>  sent!
[                 590] <s>  sent!
[                 670] <s>  sent!
[                 750] <s>  sent!
[                 830] <s>  sent!
[                 910] <s>  sent!
[                 990] <s>  sent!
[                1070] <s>  sent!
[                1150] <s>  sent!
[                1230] <s>  sent!
[                1310] <s>  sent!
[                1390] <s>  sent!
[                1470] <s>  sent!
[                1550] <s>  sent!
[                1630] <s>  sent!
[                1710] <s>  sent!
[                1790] <s>  sent!
[                1870] <s>  sent!
[                1950] <s>  sent!
[                2030] <s>  sent!
[                2110] <s>  sent!
[                2190] <s>  sent!
[                2270] <s>  sent!
[                2350] <s>  sent!
[                2430] <s>  sent!
[                2510] <s>  sent!
[                2590] <s>  sent!
[                2670] <s>  sent!
[                2750] <s>  sent!
[                2830] <s>  sent!
[                2910] <s>  sent!
[                2990] <s>  sent!
[                3070] <s>  sent!
[                3150] <s>  sent!
[                3230] <s>  sent!
[                3310] <s>  sent!
[                3390] <s>  sent!
[                3470] <s>  sent!
[                3550] <s>  sent!
[                3630] <s>  sent!
[                3710] <s>  sent!
[                3790] <s>  sent!
[                3870] <s>  sent!
[                3950] <s>  sent!
[                4030] <s>  sent!
[                4110] <s>  sent!
[                4190] <s>  sent!
[                4270] <s>  sent!
[                4350] <s>  sent!
[                4430] <s>  sent!
[                4510] <s>  sent!
[                4590] <s>  sent!
[                4670] <s>  sent!
[                4750] <s>  sent!
[                4830] <s>  sent!
[                4910] <s>  sent!
[                4990] <s>  sent!
[                5070] <s>  sent!
[                5150] <s>  sent!
[                5230] <s>  sent!
[                5310] <s>  sent!
[                5390] <s>  sent!
[                5470] <s>  sent!
[                5550] <s>  sent!
[                5630] <s>  sent!
[                5710] <s>  sent!
[                5790] <s>  sent!
[                5870] <s>  sent!
[                5950] <s>  sent!
[                6030] <s>  sent!
[                6110] <s>  sent!
[                6190] <s>  sent!
[                6270] <s>  sent!
[                6350] <s>  sent!
[                6430] <s>  sent!
[                6510] <s>  sent!
[                6590] <s>  sent!
[                6670] <s>  sent!
[                6750] <s>  sent!
[                6830] <s>  sent!
[                6910] <s>  sent!
[                6990] <s>  sent!
[                7070] <s>  sent!
[                7150] <s>  sent!
[                7230] <s>  sent!
[                7310] <s>  sent!
[                7390] <s>  sent!
[                7470] <s>  sent!
[                7550] <s>  sent!
[                7630] <s>  sent!
[                7710] <s>  sent!
[                7790] <s>  sent!
[                7870] <s>  sent!
[                7950] <s>  sent!
[                8030] <s>  sent!
